File: backend/GameApp.Domain/ValueObjects/scenes/SceneName.cs
namespace GameApp.Domain.ValueObjects.Scenes;

// Value Object of Scene Name
public class SceneName
{
    private static readonly string _messageIfEmpty = "Scene name cannot be empty";
    private readonly string Name;



    // Constructor
    public SceneName(string value)
    {
        if (string.IsNullOrWhiteSpace(value))
            throw new ArgumentException(_messageIfEmpty, nameof(value));

        Name = value;
    }

    public string GetName() => Name;


    public SceneName SetName(string newName)
    {
        if (string.IsNullOrWhiteSpace(newName))
            throw new ArgumentException(_messageIfEmpty, nameof(newName));

        return new SceneName(newName);
    }

    public override bool Equals(object? obj)
    {
        if (obj is not SceneName other)
            return false;

        return Name == other.Name;
    }

    public override int GetHashCode() => Name.GetHashCode();

    public override string ToString() => Name;
}

File: frontend/gameapp-frontend/src/domain/value-objects/scenes/scene-name.ts
// Value Object for Scene Name
export class SceneName {
    static ERROR_EMPTY = "Scene name cannot be empty";

    private _name: string;

    constructor(name: string) {
        this._name = name;
    }

    get name(): string {
        return this._name;
    }

    setName(newName: string): SceneName {
        return new SceneName(newName);
    }

    equals(other: unknown): boolean {
        return other instanceof SceneName &&
               this._name === other._name;
    }

    toString(): string {
        return this._name;
    }
}

